### PR TITLE
SDC+NSE part II

### PR DESCRIPTION
**Pull request by zingale**
This builds on #423 and completes the coupling of NSE and the simplified-SDC integration.  Previously we updated the SDC integrator code path to use the aux data for the thermodynamics.  Now we do the test on whether we are in NSE and if so, do a predictor-corrector update of the state to capture the advection + NSE changes.

### CODE REVIEW

**Comment by zingale** [comment]
This seems to work with the test_sdc unit test and is ready for review.  We will probably tweak it some in a subsequent PR.

**Comment by zingale** [comment]
when tested with Castro's nse_test convergence test, I get:
```
 $\rho$                          4.558394e+19   1.633      1.470134e+19   1.578      4.922809e+18
 $\rho u$                        1.945412e+28   1.648      6.208263e+27   1.513      2.175903e+27
 $\rho v$                        1.945412e+28   1.648      6.208263e+27   1.513      2.175903e+27
 $\rho E$                        5.709792e+37   1.620      1.857481e+37   1.552       6.33297e+36
 $\rho e$                        5.708245e+37   1.620      1.857138e+37   1.553      6.331188e+36
 $T$                              5.87327e+20   1.621      1.909652e+20   1.494      6.779334e+19
 $\rho X(\isotm{H}{1})$          4.558394e-11   1.633      1.470134e-11   1.578      4.922789e-12
 $\rho X(\isotm{He}{4})$         2.452726e+17   1.835      6.876491e+16   1.657      2.180928e+16
 $\rho X(\isotm{C}{12})$         1.130271e+12   1.846      3.145071e+11   1.678      9.830893e+10
 $\rho X(\isotm{O}{16})$         5.339327e+12   1.830      1.502096e+12   1.667      4.729501e+11
 $\rho Y_e$                      2.369726e+19   1.644      7.582117e+18   1.580      2.535685e+18
 $\rho \bar{A}$                  4.061261e+21   1.678      1.269587e+21   1.675      3.975265e+20
 $\rho (B/A)$                    4.224215e+20   1.664      1.333056e+20   1.602      4.391338e+19
```

This is better convergence than seen with Strang.  There might still be some tweaks we can do here, but this is a great start.

File: interfaces/burner.H [hunk 1]
```diff
@@ -27,31 +24,29 @@ void burner (burn_t& state, Real dt)
...
 
     } else {
         // burn as usual
         integrator(state, dt);
 
-        // update the aux from the new X's
+#ifndef SIMPLIFIED_SDC
+        // update the aux from the new X's this is not needed with the
+        // SDC method, since we took care of that internally.
         set_nse_aux_from_X(state);
```
Comment (maxpkatz) [inline interfaces/burner.H:40]: What handles this for SDC?

Comment (maxpkatz) [reply interfaces/burner.H:40]: I guess I don't understand the comment and how SDC is different from Strang for the non-NSE case.

Comment (zingale) [reply interfaces/burner.H:40]: for SDC, the solution is stored in `burn_state.y[]`, so we don't need to update `burn_state.aux[]`

**Review by harpolea — APPROVED** [review]

**Pull request merged**